documentation.md

```diff
@@ -0,0 +1,96 @@
+# Typology Processor
+
+## Overview
+
+Here's a brief explanation of each participant:
+
+- Caller: The caller initiates the request to process a transaction.
+- Handle Transaction: This function handles the transaction, coordinates the processing steps, and accumulates the results.
+- Execute Request: This function processes the transaction for each typology and sends the results to CMS and CADP.
+- Evaluate Typology Expr: This function computes the typology result value based on the provided expression.
+- Redis Cache: Redis Cache stores interim rule results for faster processing and manages cache keys.
+- Database: The Database provides the typology expression needed for evaluation.
+- CMS: The CMS receives typology results that exceed the threshold.
+- CADP: The CADP receives all the typology results.
+
+## Sequence Diagram
+
+```mermaid
+sequenceDiagram
+    autonumber
+    participant Caller as Caller
+    participant HandleTransaction as Handle Transaction
+    participant ExecuteRequest as Execute Request
+    participant EvaluateTypologyExpr as Evaluate Typology Expr
+    participant RedisCache as Redis Cache
+    participant Database as Database
+    participant CMS as CMS
+    participant CADP as CADP
+
+    Note over Caller, HandleTransaction: The process starts by handling a transaction.
+
+    Caller->>HandleTransaction: call(transaction, networkMap, ruleResult)
+    HandleTransaction->>+ExecuteRequest: call(transaction, typology, ruleResult, networkMap)
+
+    Note over ExecuteRequest: Execute Request processes the transaction for each typology.
+
+    loop channel.typologies
+        ExecuteRequest->>RedisCache: addOneGetAll(cacheKey, ruleResult)
+        ExecuteRequest->>RedisCache: check cached ruleResults
+
+        Note over RedisCache: Redis Cache stores interim rule results for faster processing.
+
+        opt no cached ruleResults
+            ExecuteRequest->>HandleTransaction: return CADPRequest
+        end
+        ExecuteRequest->>Database: getTypologyExpression(typology)
+
+        Note over Database: Database provides the typology expression needed for evaluation.
+
+        ExecuteRequest->>EvaluateTypologyExpr: call(ruleValues, ruleResults, typologyExpression)
+
+        Note over EvaluateTypologyExpr: Evaluate Typology Expr computes the typology result value.
+
+        opt typologyResultValue > expression.threshold
+            ExecuteRequest->>CMS: send Typology result
```

Comment:
How does the typology processor know where to send the result? Is this read from an environment variable? Where are those stored, and is it worth adding that to the diagram as a resource?

Reply:
Configurable endpoint in ENV variables (populated by dockerfile) - where all other config is (eg. Arango details, redis, etc).